### PR TITLE
seslib: more robust dashboard tunnelling

### PR DESCRIPTION
**Pull request by smithfarm**
We assume that "master" will be an "admin" node, and hence that
"ceph mgr services" will work there.

This commit changes the logic so the output of "ceph mgr services" is
parsed in the Python code, instead of by piping the output to "jq" and
"sed" on the cluster node like we were doing before.

It also adds a regression test for the dashboard tunnelling feature,
which also asserts that at least *some* form of dashboard downstream
branding has been applied in SES5, SES6 and SES7.

Also take the opportunity to make "contrib/standalone.sh"
ShellCheck-clean.

Fixes: https://github.com/SUSE/sesdev/issues/269
Signed-off-by: Nathan Cutler <ncutler@suse.com>

### CODE REVIEW

**Comment by smithfarm** [comment]
Interestingly enough, `sesdev tunnel DEP_ID dashboard` works on a default `ses5` deployment:

```
+ test_tunnel_dashboard ses5-4node
+ local dep_id=ses5-4node
+ sleep 30
+ timeout 40s sesdev tunnel ses5-4node dashboard
Opening tunnel to service 'dashboard' in deployment 'ses5-4node'...
Service 'dashboard' was not found in this deployment
+ curl --silent --insecure https://127.0.0.1:8443
+ grep 'SUSE Enterprise Storage'
  <title>SUSE Enterprise Storage</title>
+ sleep 15
```

File: contrib/standalone.sh [hunk 1]
```diff
@@ -139,7 +155,9 @@ if [ "$SES5" ] ; then
     run_cmd sesdev destroy --non-interactive ses5-1node
     run_cmd sesdev create ses5 --non-interactive --roles [master,client,openattic],[storage,mon,mgr,rgw],[storage,mon,mgr,mds,nfs],[storage,mon,mgr,mds,rgw,nfs] ses5-4node
     run_cmd sesdev qa-test ses5-4node
+    test_tunnel ses5-4node openattic http 8080 Redirecting
     run_cmd sesdev destroy --non-interactive ses5-4node
+    curl --silent --insecure "http://127.0.0.1:8080" || true
```
Comment (toabctl) [inline contrib/standalone.sh:160]: Why is this needed?

Comment (smithfarm) [reply contrib/standalone.sh:160]: I would like to assert that tunnel is gone after `test_tunnel` completes, but this line was just a temporary experiment. It is reworked in the commit I will be pushing presently.

File: contrib/standalone.sh [hunk 2]
```diff
@@ -63,25 +65,54 @@ function usage {
...
+    [ -z "$port" ]          && port="8443"
+    [ -z "$search_string" ] && search_string="SUSE Enterprise Storage"
+    timeout 40s sesdev tunnel "$dep_id" "$service" &
+    sleep 30
+    curl --silent --insecure "$protocol://127.0.0.1:$port" | tee "$TEMP_FILE"
+    grep "$search_string" "$TEMP_FILE"
+    sleep 15
+}
+
+function tunnel_gone {
```
Comment (toabctl) [inline contrib/standalone.sh:102]: I don't understand why we want to test this. It's testing that the `timeout` command is working and kills the process correctly. Do we want to test `timeout` here? Or if we want to test that `sesdev tunnel` is reacting to the signal correctly, can we use `kill %1` instead of `timeout` and `sleep`? That would safe us at least 15 seconds per run.
I'm also fine with just merging. Just thinking loud...

File: contrib/standalone.sh [hunk 3]
```diff
@@ -157,7 +190,9 @@ if [ "$SES6" ] ; then
     run_cmd sesdev destroy --non-interactive ses6-1node
     run_cmd sesdev create ses6 --non-interactive ses6-4node
     run_cmd sesdev qa-test ses6-4node
+    test_tunnel ses6-4node
     run_cmd sesdev destroy --non-interactive ses6-4node
+    curl --silent --insecure "https://127.0.0.1:8443" || true
```
Comment (toabctl) [inline contrib/standalone.sh:195]: shouldn't that be `tunnel_gone ...`?

File: contrib/standalone.sh (hunk 2 above)
Comment (smithfarm) [reply contrib/standalone.sh:102]: Hi Tom, hopefully https://github.com/SUSE/sesdev/issues/276 answers the question.

File: contrib/standalone.sh (hunk 3 above)
Comment (smithfarm) [reply contrib/standalone.sh:200]: good catch! thanks
